decoder/decoder.go

```diff
@@ -134,5 +150,9 @@ func (decoder *DecoderStruct) DecodePacketData(data []byte, ci *gopacket.Capture
 		}
 
 		decoder.tcpProc.Process(&decoder.tcp, &packet)
+	} else if has_icmp4 {
+		decoder.icmpProc.ProcessICMPv4(&decoder.icmp4, &packet)
+	} else if has_icmp6 {
+		decoder.icmpProc.ProcessICMPv6(&decoder.icmp6, &packet)
```

Comment:
Ok for now. But would like to see nicer way of hooking into different layers in future. E.g. for collecting advanced statistics on specific layers.